packages/trpc/server/routers/viewer/availability/schedule/update.handler.ts

```diff
@@ -38,14 +39,24 @@ export const updateHandler = async ({ input, ctx }: UpdateOptions) => {
     },
   });
 
-  if (userSchedule?.userId !== user.id) throw new TRPCError({ code: "UNAUTHORIZED" });
-
-  if (!userSchedule || userSchedule.userId !== user.id) {
+  if (!userSchedule) {
     throw new TRPCError({
       code: "UNAUTHORIZED",
     });
   }
 
+  if (userSchedule?.userId !== user.id) {
+    const hasEditPermission = await hasEditPermissionForUserID({
+      ctx,
+      input: { memberId: userSchedule.userId },
+    });
+    if (!hasEditPermission) {
+      throw new TRPCError({
+        code: "UNAUTHORIZED",
+      });
+    }
+  }
```

Comment:
If the schedule ID doesnt match authedUID 

Check if the UID has permission to edit this schedule